rustls/src/stream.rs

```diff
@@ -41,15 +41,8 @@ where
 
         Ok(())
     }
-}
 
-impl<'a, C, T, S> Read for Stream<'a, C, T>
-where
-    C: 'a + DerefMut + Deref<Target = ConnectionCommon<S>>,
-    T: 'a + Read + Write,
-    S: SideData,
-{
-    fn read(&mut self, buf: &mut [u8]) -> Result<usize> {
+    fn prepare_read(&mut self) -> Result<()> {
```

Comment:
I'd like to see a separate commit for the extraction of `prepare_read()`, too. This commit ("Implement BufRead for Stream/StreamOwned") has too much going on right now.